x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -831,30 +896,38 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static class ReactiveReason implements AutoscalingDeciderResult.Reason {
 
         static final int MAX_AMOUNT_OF_SHARDS = 512;
+        static final int MAX_AMOUNT_OF_DECISIONS = 512;
```

Comment:
This means we get a list of decisions that is

1. Fairly large.
2. Not covering shards in total, i.e., the last shard will have some decisions.
3. Does not include the shard-id in the decision.

I think we should move to a Map of shard-id to decision list(s). And then limit the number of shards to explain to something like 5.